PIMTool/Apis/WebAPI/DependencyInjection.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Application.Interfaces;
+using FluentValidation.AspNetCore;
+using System.Diagnostics;
+using WebAPI.Middlewares;
+
+namespace WebAPI
+{
+    public static class DependencyInjection
+    {
+        public static IServiceCollection AddWebAPIService(this IServiceCollection services)
+        {
+            services.AddControllers();
+            services.AddEndpointsApiExplorer();
+            services.AddSwaggerGen();
+            services.AddHealthChecks();
+            services.AddSingleton<GlobalExceptionMiddleware>();
```

Comment:
[C] it was bad code to use AddSingleton for middleware .
Middleware did not need to declare singleton
The same for StopWatch.
For case one million requrest paralel, stopwatch singleton is bad design